src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -306,6 +310,20 @@ private function registerBusMiddleware(ContainerBuilder $container, string $busI
                 throw new RuntimeException(sprintf('Invalid middleware factory "%s": a middleware factory must be an abstract definition.', $id));
             }
 
+            if ($debug) {
+                $container->register($debugMiddlewareId = "debug.traced.$messengerMiddlewareId", TraceableMiddleware::class)
```

Comment:
I would use `messenger.debug.traced.` as prefix, to ensure that we don't get conflicts with other components wanting to do debug things in the future.

Reply:
and we might want to make it hidden